addons/sales_team/models/res_users.py

```diff
@@ -16,7 +16,7 @@ def create(self, vals):
         # Assign the new user in the sales team if there's only one sales team of type `Sales`
         user = super(ResUsers, self).create(vals)
         if user.has_group('sales_team.group_sale_salesman') and not user.sale_team_id:
-            teams = self.env['crm.team'].search([('team_type', '=', 'sales')])
+            teams = self.env['crm.team'].search([(1, '=', 1)])
```

Comment:
This will cost a search at each user creation, and it will only be usefull in the case we got only one team in the database. Isn't there a domain we can put here ?
Or, ask your PO if it is still required to put the user in a team at its creation.